src/net_processing.cpp

```diff
@@ -5841,7 +5873,35 @@ bool PeerManagerImpl::SendMessages(CNode* pto)
                         }
                         if (tx_relay->m_bloom_filter && !tx_relay->m_bloom_filter->IsRelevantAndUpdate(*txinfo.tx)) continue;
                         // Send
-                        vInv.push_back(inv);
+                        bool fanout = true;
+                        const auto wtxid = txinfo.tx->GetWitnessHash();
+                        if (reconciles_txs) {
+                            auto txiter = m_mempool.GetIter(txinfo.tx->GetHash());
+                            if (txiter) {
+                                if ((*txiter)->GetCountWithDescendants() > 1) {
+                                    // If a transaction has in-mempool children, always fanout it.
+                                    // Until package relay is implemented, this is needed to avoid
+                                    // breaking parent+child relay expectations in some cases.
+                                    //
+                                    // Potentially reconciling parent+child would mean that for every
+                                    // child we need to to check if any of the parents is currently
+                                    // reconciled so that the child isn't fanouted ahead. But then
+                                    // it gets tricky when reconciliation sets are full: a) the child
+                                    // can't just be added; b) removing parents from reconciliation
+                                    // sets for this one child is not good either.
+                                    fanout = true;
```

Comment:
Since `fanout` is already initialized `true`, couldn't we simplify it?

```diff
@@ -5878,19 +5878,17 @@ bool PeerManagerImpl::SendMessages(CNode* pto)
                         if (reconciles_txs) {
                             auto txiter = m_mempool.GetIter(txinfo.tx->GetHash());
                             if (txiter) {
-                                if ((*txiter)->GetCountWithDescendants() > 1) {
-                                    // If a transaction has in-mempool children, always fanout it.
-                                    // Until package relay is implemented, this is needed to avoid
-                                    // breaking parent+child relay expectations in some cases.
-                                    //
-                                    // Potentially reconciling parent+child would mean that for every
-                                    // child we need to to check if any of the parents is currently
-                                    // reconciled so that the child isn't fanouted ahead. But then
-                                    // it gets tricky when reconciliation sets are full: a) the child
-                                    // can't just be added; b) removing parents from reconciliation
-                                    // sets for this one child is not good either.
-                                    fanout = true;
-                                } else {
+                                // If a transaction has in-mempool children, always fanout it.
+                                // Until package relay is implemented, this is needed to avoid
+                                // breaking parent+child relay expectations in some cases.
+                                //
+                                // Potentially reconciling parent+child would mean that for every
+                                // child we need to to check if any of the parents is currently
+                                // reconciled so that the child isn't fanouted ahead. But then
+                                // it gets tricky when reconciliation sets are full: a) the child
+                                // can't just be added; b) removing parents from reconciliation
+                                // sets for this one child is not good either.
+                                if ((*txiter)->GetCountWithDescendants() <= 1) {
                                     auto fanout_randomizer = m_connman.GetDeterministicRandomizer(RANDOMIZER_ID_FANOUTTARGET);
                                     fanout = m_txreconciliat
ion->ShouldFanoutTo(wtxid, fanout_randomizer, pto->GetId(),
                                                             
                    inbounds_nonrcncl_tx_relay, outbounds_non
rcncl_tx_relay);
```